sugondat/chain/node/src/proposer.rs

```diff
@@ -89,8 +90,58 @@ impl<P: ProposerInterface<Block> + Send> ProposerInterface<Block> for BlockLimit
             // testing for detection of healthiness.
             parent_header.number == 0
         };
+        let exceeded_max_skipped_blocks = {
+            let maybe_last_relay_block_number: Option<u32> =
+                inherent_digests.logs.iter().find_map(|log| {
```

Comment:
This approach isn't going to work. These inherent digests are produced by the consensus engine and are fed _into_ the runtime, whereas you are looking for a digest item produced _by_ the runtime.

What should be done instead is:
  1. Get a timestamp or relay parent number from the `parent_header`. This could be done in a few different ways, discussed further on.
  2. Compare that to the current timestamp or relay parent number.

When it comes to extracting a timestamp, I see the following possibilities:
  1. Read the state at the parent block directly. The state is available - so we can just read a `well_known_key`.
  1. Scrape the inherents from the block body. Either the timestamp inherent or the parachain inherent.
  1. Introduce a digest to each header which contains the relay parent number. Actually, this is already done (https://github.com/paritytech/polkadot-sdk/blob/bbfff668437245690b4752fb0dc0c8d9eb1b1e6e/cumulus/primitives/core/src/lib.rs#L293). Though at some point it will be removed in favor of the relay parent hash.

I would prefer the non-invasive options (1) and (2) rather than introduce a new digest, as that adds additional bloat to the chain and eats into the PoV space, albeit very slightly.